pkg/crypto/walletapi/wallet_api.go

```diff
@@ -56,6 +65,33 @@ type Crypto interface {
 	//
 	//		in case of error, the index will be -1
 	FindVerKey(candidateKeys []string) (int, error)
+
+	// ConvertToEncryptionKey will create and persist a Curve25519 encryption keypair
+	// created from the ed25519 keypair with the provided public key.
+	//
+	// Returns:
+	//
+	// []byte: The public key of the encryption keypair created. nil in case of error.
+	//
+	// error:
+	ConvertToEncryptionKey(pubkey []byte) ([]byte, error)
+
+	// AttachCryptoOperator attaches a crypto operator to this wallet, so the operator can use its private keys.
+	AttachCryptoOperator(cryptoOp KeyHolderUser) error
+}
+
+// KeyHolderUser denotes an object that accesses keys in a KeyHolder
+// A KeyHolderUser must be defined inside pkg/crypto to fulfill the
+// interface, which ensures that access to private keys is limited
+// to only packages in pkg/crypto
+type KeyHolderUser interface {
+
+	// InjectKeyHolder injects a KeyHolder which will provide private keys for the KeyHolderUser's crypto computations.
+	// Usage note: InjectKeyHolder is not for the client to call,
+	// the client can't access the secret wallet (KeyHolder) to pass in.
+	// This is used by a wallet implementing the Crypto interface,
+	// in AttachCryptoOperator(KeyHolderUser)
+	InjectKeyHolder(secretapi.KeyHolder) error
```

Comment:
Should we expose this interface in walletapi considering that client should not call this interface? Can you declare it in receiver only?

Reply:
I'm still stuck on this - for the mock wallet to have the correct overrides, it needs an AttachCryptoOperator signature that has a `KeyHolderUser` (which I've renamed to CryptoOperator) as a parameter. If `KeyHolderUser` is internal or consumer-only, then when I mock it inside `mock/wallet` the mockwallet `AttachCryptoOperator()` method has a different signature to the method in the `Crypto` interface, so `mockwallet.CloseableWallet` doesn't implement the wallet interfaces.